src/components/sideMenu/view/sideMenuView.js

```diff
@@ -20,53 +20,46 @@ import styles from './sideMenuStyles';
 // Images
 const SIDE_MENU_BACKGROUND = require('../../../assets/side_menu_background.png');
 
-class SideMenuView extends Component {
-  /* Props
-   * ------------------------------------------------
-   *   @prop { type }    name                - Description....
-   */
-
-  constructor(props) {
-    super(props);
-    this.state = {
-      menuItems: props.isLoggedIn ? MENU.AUTH_MENU_ITEMS : MENU.NO_AUTH_MENU_ITEMS,
-      isAddAccountIconActive: false,
-      storageT: 'R',
-    };
-  }
+const SideMenuView = ({
+  currentAccount,
+  isLoggedIn,
+  intl,
+  handleLogout,
+  accounts,
+  switchAccount,
+  navigateToRoute,
+}) => {
+  const [menuItems, setMenuItems] = useState(
+    isLoggedIn ? MENU.AUTH_MENU_ITEMS : MENU.NO_AUTH_MENU_ITEMS,
+  );
+  const [isAddAccountIconActive, setIsAddAccountIconActive] = useState(false);
+  const [storageT, setStorageT] = useState('R');
 
   // Component Life Cycles
-  componentDidMount() {
+  useEffect(() => {
+    let _isMounted = false;
     getStorageType().then(item => {
-      this.setState({ storageT: item });
+      if (!_isMounted) {
+        setStorageT(item);
+      }
     });
-  }
+  }, []);
 
   // Component Functions
 
-  _handleOnPressAddAccountIcon = () => {
-    const { isLoggedIn, accounts } = this.props;
-    const { isAddAccountIconActive } = this.state;
-
+  const _handleOnPressAddAccountIcon = () => {
     if (!isAddAccountIconActive) {
-      this.setState({ menuItems: accounts });
+      setMenuItems(accounts);
     } else {
-      this._setMenuItems(isLoggedIn);
+      setMenuItems(isLoggedIn ? MENU.AUTH_MENU_ITEMS : MENU.NO_AUTH_MENU_ITEMS);
     }
 
-    this.setState({ isAddAccountIconActive: !isAddAccountIconActive });
-  };
-
-  _setMenuItems = isLoggedIn => {
-    this.setState({
-      menuItems: isLoggedIn ? MENU.AUTH_MENU_ITEMS : MENU.NO_AUTH_MENU_ITEMS,
-    });
+    setIsAddAccountIconActive(!isAddAccountIconActive);
   };
 
-  _handleOnMenuItemPress = item => {
-    const { navigateToRoute, switchAccount } = this.props;
-
+  const _handleOnMenuItemPress = item => {
     if (item.id === 'logout') {
+      // eslint-disable-next-line react/no-this-in-sfc
       this.ActionSheet.show();
```

Comment:
you cannot use this keyword outside of react.component. if you want to ref with react hooks, it has useRef() func. if you can search on all repo you can see usage

Reply:
ok!